stdlib/regex.gr

```diff
@@ -1921,155 +2375,418 @@ let subArraysEqual = (arr1, start1, arr2, start2, length) => {
+    ) => {
+    if ({
+      pos + len <= limit && subArraysEqual(
+        buf.matchInputExploded,
+        pos,
+        String.explode(toMatch),
+        0,
+        len,
+      )
+    }) {
+      next_m(buf, pos + len, start, limit, end, state, stack)
+    } else {
+      None
+    }
+  }
 
-let stringMatcherIterated = (toMatch, len, max) => iterateMatcher((buf: MatchBuf, pos: Number, start: Number, limit: Number, end: Number, state, stack) => {
-  subArraysEqual(buf.matchInputExploded, pos, String.explode(toMatch), 0, len)
-}, len, max)
+let stringTailMatcher = (toMatch, len) =>
+
+    (
+      buf: MatchBuf,
+      pos: Number,
+      start: Number,
+      limit: Number,
+      end: Number,
+      state,
+      stack,
+    ) => {
+    if ({
+      pos + len <= limit && subArraysEqual(
+        buf.matchInputExploded,
+        pos,
+        String.explode(toMatch),
+        0,
+        len,
+      )
+    }) {
+      Some(pos + len)
+    } else {
+      None
+    }
+  }
 
+let stringMatcherIterated = (toMatch, len, max) =>
+  iterateMatcher(
+
+      (
+        buf: MatchBuf,
+        pos: Number,
+        start: Number,
+        limit: Number,
+        end: Number,
+        state,
+        stack,
+      ) => {
+      subArraysEqual(
+        buf.matchInputExploded,
+        pos,
+        String.explode(toMatch),
+        0,
+        len,
+      )
+    },
+    len,
+    max,
+  )
 
 // match nothing
 
-let neverMatcher = (buf: MatchBuf, pos: Number, start: Number, limit: Number, end: Number, state, stack) => {
+let neverMatcher =
+  (
+    buf: MatchBuf,
+    pos: Number,
+    start: Number,
+    limit: Number,
+    end: Number,
+    state,
+    stack,
+  ) => {
   None
 }
 
 // match any byte
 
-let anyMatcher = (next_m) => (buf: MatchBuf, pos: Number, start: Number, limit: Number, end: Number, state, stack) => {
-  if ({
-    pos < limit
-  }) next_m(buf, pos + 1, start, limit, end, state, stack) else None
-}
+let anyMatcher = next_m =>
+
+    (
+      buf: MatchBuf,
+      pos: Number,
+      start: Number,
+      limit: Number,
+      end: Number,
+      state,
+      stack,
+    ) => {
+    if ({
+      pos < limit
+    }) {
+      next_m(buf, pos + 1, start, limit, end, state, stack)
+    } else {
+      None
+    }
+  }
 
-let anyTailMatcher = () => (buf: MatchBuf, pos: Number, start: Number, limit: Number, end: Number, state, stack) => {
-  if ({
-    pos < limit
-  }) Some(pos + 1) else None
-}
+let anyTailMatcher = () =>
+
+    (
+      buf: MatchBuf,
+      pos: Number,
+      start: Number,
+      limit: Number,
+      end: Number,
+      state,
+      stack,
+    ) => {
+    if ({
+      pos < limit
+    }) {
+      Some(pos + 1)
+    } else {
+      None
+    }
+  }
 
-let anyMatcherIterated = (max) => (buf: MatchBuf, pos: Number, start: Number, limit: Number, end: Number, state, stack) => {
-  let n = match(max) {
-    None => limit - pos,
-    Some(max) => min(max, limit - pos),
+let anyMatcherIterated = max =>
+
+    (
```

Comment:
Looks like we got an extra space here.

Reply:
Nice catch - it's when a function returns a lambda.  Fixed.